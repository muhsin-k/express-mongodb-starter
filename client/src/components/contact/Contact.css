.contact {
  border: 1px solid black;
  padding: 8px;
  height: 120px; }

p {
  margin: 0px; }

.button-container {
  padding: 0px;
  padding-left: 0px !important; }

.main-area {
  height: 70%; }
