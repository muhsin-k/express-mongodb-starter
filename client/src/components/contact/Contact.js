import React, { Component } from "react";
import PropTypes from "prop-types";
import DefaultImage from "../../assets/default-image.png";
import "./Contact.css";
class Contact extends Component {
  deleteContact() {
    this.props.deleteContact(this.props);
  }
  editContact() {
    this.props.editContact(this.props);
  }
  viewContact() {
    this.props.viewContact(this.props);
  }
  renderProfilePic() {
    if (this.props.profilePic) {
      return (
        <img
          src={this.props.profilePic}
          width="100"
          height="100"
          alt="profilepic"
        />
      );
    }
    return <img src={DefaultImage} width="100" height="100" alt="profilepic" />;
  }
  renderActionButtons() {
    if (this.props.showActions) {
      return (
        <div className="row">
          <div className="col s4 button-container">
            <a
              className="waves-effect waves-light btn"
              onClick={this.viewContact.bind(this)}
            >
              <i className="material-icons left">info</i>
            </a>
          </div>

          <div className="col s4 button-container">
            <a
              className="waves-effect waves-light btn"
              onClick={this.editContact.bind(this)}
            >
              <i className="material-icons left">edit</i>
            </a>
          </div>
          <div className="col s4 button-container">
            <a
              className="waves-effect waves-light btn"
              // onClick={props.deleteContact(props)}
              onClick={this.deleteContact.bind(this)}
            >
              <i className="material-icons left">delete</i>
            </a>
          </div>
        </div>
      );
    } else {
      return <div />;
    }
  }
  render() {
    return (
      <div className="contact">
        <div className="row main-area ">
          <div className="col s4">{this.renderProfilePic()}</div>
          <div className="col s8">
            <div className="row">
              <p>
                {this.props.firstName} {this.props.secondName}
              </p>
              <p>{this.props.email} </p>
            </div>
            {this.renderActionButtons()}
          </div>
        </div>
      </div>
    );
  }
}
Contact.propTypes = {
  title: PropTypes.string,
  firstName: PropTypes.string,
  secondName: PropTypes.string
};

export default Contact;
