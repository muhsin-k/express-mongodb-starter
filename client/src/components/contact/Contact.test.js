import React from "react";
import ReactDOM from "react-dom";
import * as enzyme from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Contact from "./Contact";

enzyme.configure({
  adapter: new Adapter()
});

describe("Header", () => {
  it("renders without crashing", () => {
    enzyme.mount(<Contact />);
  });

  it("should render correctly[Snapshot]", () => {
    const output = enzyme.shallow(
      <Contact
        firstName="Muhsin"
        secondName="Keloth"
        email="muhsinkeramam@gmail.com"
        id="5ae17aed0299c3a5a76018a6"
        profilePic="https://d2ojpxxtu63wzl.cloudfront.net/static/40d467520759865d5b2b669e242033f5_08fbfcfd9529b0affb3efe7090dee8fb3b92a494cb83b76d76c8e1130472a642"
        showActions={true}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
