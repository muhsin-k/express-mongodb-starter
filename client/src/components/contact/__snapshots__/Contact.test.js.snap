// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly[Snapshot] 1`] = `
<div
  className="contact"
>
  <div
    className="row main-area "
  >
    <div
      className="col s4"
    >
      <img
        alt="profilepic"
        height="100"
        src="https://d2ojpxxtu63wzl.cloudfront.net/static/40d467520759865d5b2b669e242033f5_08fbfcfd9529b0affb3efe7090dee8fb3b92a494cb83b76d76c8e1130472a642"
        width="100"
      />
    </div>
    <div
      className="col s8"
    >
      <div
        className="row"
      >
        <p>
          Muhsin
           
          Keloth
        </p>
        <p>
          muhsinkeramam@gmail.com
           
        </p>
      </div>
      <div
        className="row"
      >
        <div
          className="col s4 button-container"
        >
          <a
            className="waves-effect waves-light btn"
            onClick={[Function]}
          >
            <i
              className="material-icons left"
            >
              info
            </i>
          </a>
        </div>
        <div
          className="col s4 button-container"
        >
          <a
            className="waves-effect waves-light btn"
            onClick={[Function]}
          >
            <i
              className="material-icons left"
            >
              edit
            </i>
          </a>
        </div>
        <div
          className="col s4 button-container"
        >
          <a
            className="waves-effect waves-light btn"
            onClick={[Function]}
          >
            <i
              className="material-icons left"
            >
              delete
            </i>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
